src/sentry/utils/contexts_normalization.py

```diff
@@ -66,3 +68,90 @@ def normalize_runtime(data):
             version = version_map.get(build, None)
             if version is not None:
                 data['version'] = version
+
+
+def _get_version(user_agent):
+    return '.'.join(
+        value for value in [
+            user_agent['major'],
+            user_agent['minor'],
+            user_agent.get('patch'),
+        ] if value
+    ) or None
+
+
+def _parse_user_agent(data):
+    http = data.get('request')
+    if not http:
+        return None
+
+    headers = http.get('headers')
+    if not headers:
+        return None
+
+    try:
+        for key, value in headers:
+            if key != 'User-Agent':
```

Comment:
Does the request object normalize header names to this casing?

Reply:
Yap and it's one of the things that's pretty annoying because it picks the wrong normalization. Modern HTTP requires lowercase :(

Reply:
Good point, not sure, should I change this now?